core/src/main/java/com/linecorp/armeria/client/endpoint/dns/DnsEndpointGroupBuilder.java

```diff
@@ -153,6 +157,29 @@ final DefaultDnsResolver buildResolver(Consumer<DnsNameResolverBuilder> customiz
                                      hostsFileEntriesResolver());
     }
 
+    /**
+     * Adds the {@link DnsQuestionListener}s that listens to the result of querying {@link DnsRecord}s.
+     */
+    public DnsEndpointGroupBuilder addDnsQuestionListeners(Iterable<DnsQuestionListener> dnsQuestionListeners) {
```

Comment:
Please add `@UnstableApi` to the new public APIs.
```suggestion
    @UnstableApi
    public DnsEndpointGroupBuilder addDnsQuestionListeners(Iterable<DnsQuestionListener> dnsQuestionListeners) {
```